scripts/custom_lint_checks.py

```diff
@@ -73,17 +74,77 @@ def visit_call(self, node):
         Args:
             node. Call. The current function call node.
         """
-        if (isinstance(node.func, astroid.Name) and
+        if (isinstance(node.func, astroid.Name)) and (
                 node.func.name == self._function_name):
-            if (node.args is not None and
+            if (node.args is not None) and (
                     len(node.args) > self._positional_arguments_count):
                 self.add_message('non-explicit-kwargs', node=node)
 
 
+class HangingIndentChecker(BaseChecker):
+    """Custom pylint checker which checks for break after paren in case
+    of hanging indentation.
+    """
+    __implements__ = IRawChecker
+
+    name = 'hanging-indent'
+    priority = -1
+    msgs = {
+        'C0002': (
+            (
+                'There should be a break after parenthesis when content within '
+                'paren spans to multiple lines.'),
+            'break-after-hanging-indent',
+            (
+                'If something within parenthesis extends along multiple lines, '
+                'break after opening paren.')
+        ),
+    }
+
+    def process_module(self, node):
+        """Process a module.
+
+        Args:
+            node: Node to access module content.
+        """
+        file_content = node.stream().readlines()
+        file_length = len(file_content)
+        exclude = False
+        for line_num in range(file_length):
```

Comment:
Can you try out enumerate here instead of range? Please see the time taken by both and chose the efficient one.

Reply:
I checked for both of these and range is more efficient than enumerate (https://stackoverflow.com/questions/11990105/rangelenlist-or-enumeratelist)

Reply:
I think this was mentioned in the StackOverflow answer which you had earlier pointed to. I suggest you do a `time.time()` and find out the time taken by each of the three approaches here for a single file. Thanks!

Reply:
Oh my bad, I will do that, Thanks!

Reply:
For xrange: 0.000409126281738
For range: 0.000427007675171
For enumerate: 0.000483989715576

Reply:
Let's go with `xrange` then. Thanks for the info!